execution-core/src/transfer/phoenix.rs

```diff
@@ -658,6 +671,19 @@ impl Payload {
             0 => None,
             1 => Some(ContractExec::Call(ContractCall::from_slice(buf)?)),
             2 => Some(ContractExec::Deploy(ContractDeploy::from_slice(buf)?)),
+            3 => {
+                if buf.len() != 512 {
+                    return Err(BytesError::BadLength {
+                        found: buf.len(),
+                        expected: 512,
+                    });
+                }
```

Comment:
I'm not a fan in having this length always 512.
Any reason to not having a var len with a max of 512?

Reply:
You mean something like a `Vec<u8>`, that I then constrain during serialization? Can do. A bit more complicated, but easy enough.

Reply:
Explored it a little, and this would mean that I'd need to add an `Error` variant, and make Moonlight transaction creation possibly fail (Phoenix ones already do). Will make the change, but it'll take some time.